hoptimator-kafka-adapter/src/main/java/com/linkedin/hoptimator/catalog/kafka/KafkaTopic.java

```diff
@@ -8,14 +8,11 @@
 
 public class KafkaTopic extends Resource {
   public KafkaTopic(String name, Integer numPartitions, Map<String, String> clientOverrides) {
-    super(name, "KafkaTopic");
+    super("KafkaTopic");
+    export("name", name);
     export("numPartitions", Optional.ofNullable(numPartitions)
       .map(x -> Integer.toString(x)).orElse("null"));
     export("clientOverrides", clientOverrides);
```

Comment:
Interesting. It works, but I guess only cuz it renders as a null map, instead of an empty map. Basically, `clientOverrides: null` instead of `clientOverrides: {}`. This doesn't matter since the CRD doesn't require `clientOverrides` to be defined (nor should it), and the operators handle nulls. But I could see this being a problem for some future CRD. Hmm.

Reply:
Does that mean that for the case of an empty map, it would render `null`? 

I can make my issue clearer too. It isn't clear that `null` is not safe to pass to `export`. Or that the rendered template is valid if the map is empty. Because the yaml must have a value to the right of `:` the value must be one of the following: `null`, `{}` (empty map), or a valid map with keys. If `null` and empty are equivalent, then I think that a very "permissive" renderer is fine.

Reply:
Yes, at least so far, I'm treating `null` and `{}` the same.

Reply:
I'll leave this hanging for now.